scripting/include/srccoop/blackmesa/entitypatch.inc

```diff
@@ -791,4 +793,341 @@ public Action Timer_SetTripmineSpriteColor(Handle timer, CSpriteTeam pSpriteTeam
 	pSpriteTeam.SetRenderAlpha(0);
 
 	return Plugin_Continue;
+}
+
+//------------------------------------------------------
+// weapon_357/weapon_crossbow
+// New constructed logic of toggling ironsights.
+//------------------------------------------------------
+public MRESReturn Hook_ToggleIronsights(int _this, Handle hParams)
+{
+	if (CoopManager.IsCoopModeEnabled() && CoopManager.IsFeatureEnabled(FT_SP_WEAPONS))
```

Comment:
IsCoopModeEnabled is redundant if you check for feature

Reply:
Removed all instances of `CoopManager.IsCoopModeEnabled()` with the feature.